### DISCUSSION TITLE
NLP for Code Summary and Question Answering

### DISCUSSION THREAD

**Discussion by KevinShuman** [microsoft/vscode-discussions#551] [Extension Show and Tell]
While I do not have the time nor expertise to implement this myself, I figured I would put this idea out into the community for anyone to take up and make this a place for any discussion or questions.

I was give some code today in a language I do not know, that was badly written, and with very poor documentation and was asked to utilize it within the work I am doing. Typically, I would have run down the rabbit hole of understanding the code in detail before putting into use.

After doing this for a little bit and getting a little frustrated, I came up with another plan. I put the code into ChatGPT and asked for a summary of what's going on. I then was able to ask more pointed questions until I got a good sense of how the code was designed and how I should use it.

I figure if a VS Code extension capable of something similar could be created, that folks might find it very useful.

**Comment by olguzzar** [comment]
This is something that Copilot Chat can help with! There is a waitlist to get access, but check out [this video](https://www.youtube.com/watch?v=gDJzr9DBKTI) to see it in action.

**Reply by KevinShuman** [reply]
This looks exactly like what I was thinking. Jumped onto the wait list! Thank you!
